gm4_speed_paths/overlay_26/data/gm4_speed_paths/functions/apply_effects/add_speed.mcfunction

```diff
@@ -0,0 +1,11 @@
+# applies the Speed Paths attribute and gm4_on_path tag
+# @s = player upon stepping on a speed path
+# at @s
+# run from main
+
+# guard return
+execute unless block ~ ~-0.9 ~ minecraft:dirt_path run return fail
+
+tag @s add gm4_on_path
+scoreboard players set @s gm4_speed_paths 2
```

Comment:
Why do you need both a tag and a score to store the state?

Reply:
Following up your question in main, is it better to check @a[tag= or @a[scores ?   If tag is not better, then we can use scores exclusively

Reply:
I'm pretty confindent that the difference is negligible.